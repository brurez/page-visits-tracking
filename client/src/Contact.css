.Contact-field {
    border: 1px solid rgba(34, 36, 38, 0.15);
    border-radius: 0.28571429rem;
    padding: 0.67857143em 1em;
    line-height: 1.21428571em;
    margin: 0;
    display: inline-block;
    width: 100%;
}
