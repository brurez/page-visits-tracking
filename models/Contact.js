const mongoose = require('mongoose');
const { Schema } = mongoose;

const contactSchema = new Schema({
  name: { type: String, required: true },
  email: { type: String, required: true },
  visits: [
    {
      url: { type: String, required: true },
      visitedAt: { type: Date, required: true }
    }
  ],
  createdAt: { type: Date, default: new Date() }

});

mongoose.model('contacts', contactSchema);
