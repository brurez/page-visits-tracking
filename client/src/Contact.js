import React, { Component } from 'react';
import axios from 'axios';
import moment from 'moment';

import './Contact.css';

class Contact extends Component {
  state = { data: {} };

  componentDidMount() {
    axios
      .get(`/api/contacts/${this.props.match.params.id}`)
      .then(res => {
        //debugger;
        const { data, count } = res.data;
        this.setState({ data: data[0] });
      })
      .catch(err => console.log(err));
  }

  renderVisits() {
    if (this.state.data.visits && this.state.data.visits.length) {
      const visits = this.state.data.visits.map(visit => {
        return (
          <tr>
            <td>{moment(visit.visitedAt).format('LLL')}</td>
            <td>{visit.url}</td>
          </tr>
        );
      });

      return (
        <table className="ui celled table">
          <thead>
            <tr>
              <th>Data da Visita</th>
              <th>Endereço da Página</th>
            </tr>
          </thead>
          <tbody>{visits}</tbody>
        </table>
      );
    }
  }

  render() {
    const { name, email } = this.state.data;

    return (
      <div>
        <h2>Detalhes do Contato</h2>
        <div className="ui vertical segment">
          <div className="ui form">
            <div className="two fields">
              <div className="field">
                <label>Nome</label>
                <div className="Contact-field">{name}</div>
              </div>
              <div className="field">
                <label>Email</label>
                <div className="Contact-field">{email}</div>
              </div>
            </div>
          </div>
        </div>
        <div className="ui vertical segment">
          <h3>Visitas</h3>
          {this.renderVisits()}
        </div>
      </div>
    );
  }
}

export default Contact;
