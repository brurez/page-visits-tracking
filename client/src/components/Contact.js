import React, { Component } from 'react';
import axios from 'axios';
import moment from 'moment';
import socket from '../socket';

import './Contact.css';
import LoadingMsg from './LoadingMsg';

class Contact extends Component {
  constructor(props){
    super(props);
    this.state = { data: {}, isFetching: true };
    this.fetchApi = this.fetchApi.bind(this)
  }

  componentDidMount() {
    this.fetchApi();
    socket.emit('join', 'contact-' + this.props.match.params.id);
    socket.on('new-contact-visit', this.fetchApi);
  }

  fetchApi() {
    axios
      .get(`/api/contacts/${this.props.match.params.id}`)
      .then(res => {
        this.setState({ data: res.data, isFetching: false });
      })
      .catch(err => console.log(err));
  }

  componentWillUnmount(){
    socket.emit('leave', 'contact-' + this.props.match.params.id);
    socket.off('new-contact-visit', this.fetchApi);
  }

  renderVisits() {
    if (!this.state.isFetching) {
      const visits = this.state.data.visits.map(visit => {
        return (
          <tr key={visit._id}>
            <td>{moment(visit.visitedAt).format('LLL')}</td>
            <td>{visit.url}</td>
          </tr>
        );
      });

      return (
        <table className="ui compact celled table" id="visits">
          <thead>
            <tr>
              <th>Data da Visita</th>
              <th>Endereço da Página</th>
            </tr>
          </thead>
          <tbody>{visits}</tbody>
        </table>
      );
    } else {
      return <LoadingMsg />;
    }
  }

  render() {
    const { name, email } = this.state.data;

    return (
      <div>
        <h2>
          <i className="user icon" /> Detalhes do Contato
        </h2>
        <div className="ui vertical segment">
          <div className="ui form">
            <div className="two fields">
              <div className="field">
                <label>Nome</label>
                <div className="Contact-field" id="name-field">
                  {name}
                </div>
              </div>
              <div className="field">
                <label>Email</label>
                <div className="Contact-field" id="email-field">
                  {email}
                </div>
              </div>
            </div>
          </div>
        </div>
        <div className="ui vertical segment">
          <h3>Visitas</h3>
          {this.renderVisits()}
        </div>
      </div>
    );
  }
}

export default Contact;
