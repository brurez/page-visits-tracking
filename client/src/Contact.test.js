import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Contact from './Contact';

describe('Home Component', () => {
  test('match the snapshot', () => {
    const component = renderer.create(
      <Contact match={{ params: { id: 123 } }} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
