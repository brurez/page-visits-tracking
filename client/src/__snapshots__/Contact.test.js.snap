// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component match the snapshot 1`] = `
<div>
  <h2>
    <i
      className="user icon"
    />
     Detalhes do Contato
  </h2>
  <div
    className="ui vertical segment"
  >
    <div
      className="ui form"
    >
      <div
        className="two fields"
      >
        <div
          className="field"
        >
          <label>
            Nome
          </label>
          <div
            className="Contact-field"
          />
        </div>
        <div
          className="field"
        >
          <label>
            Email
          </label>
          <div
            className="Contact-field"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ui vertical segment"
  >
    <h3>
      Visitas
    </h3>
    <div
      className="ui icon message"
    >
      <i
        className="notched circle loading icon"
      />
      <div
        className="content"
      >
        <div
          className="header"
        >
          Espere um segundo
        </div>
        <p>
          Estamos baixando o conteúdo para você.
        </p>
      </div>
    </div>
  </div>
</div>
`;
